src/plone/app/transmogrifier/uidupdater.py

```diff
@@ -59,6 +63,11 @@ def __iter__(self):
                         obj._setUID(uid)
 
             if IAttributeUUID.providedBy(obj):
+                existing_uid = obj.portal_catalog({'UID': uid})
+                if bool(existing_uid):
+                    logger.debug('When updating UID for %s, another item with the same UID %s was found here: %s' % (
```

Comment:
sorry, I meant `logger.warn()` instead.

also, I don't know if using acquisition is better/faster than fetching the `portal_catalog` the standard way. do you mind to comment on this?

Reply:
catalog: you are right - I have changed it so its similar to plone/app/transmogrifier/reindexobject.py